src/module/output_device.rs

```diff
@@ -94,8 +96,18 @@ impl Module for OutputDevice {
                 });
 
             if let Some(output_device) = output_device {
-                let config = output_device.default_output_config()
-                    .expect("default_output_format");
+                let sample_rate = cpal::SampleRate(SAMPLE_RATE as u32);
+
+                let supported_configs = output_device
+                    .supported_output_configs()
+                    .expect("device.configs");
+
+                let config = supported_configs
+                    .filter(|config_range| config_range.sample_format() == SampleFormat::F32)
```

Comment:
`SampleFormat::F32` could stand to be matched against `crate::engine::Sample` type instead of hard-coded.